 <!-- This adds support for mkdocs-git-authors-plugin to mkdocs-material theme --> <!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><link rel=canonical href=https://pypi.org/project/pear-ebi/0.1.1.10/tutorials/1How_to_use_pear_ebi.html><link rel=prev href=../modules/test/test.html><link rel=next href=2Advanced%20use%20of%20pear_ebi.html><link rel=icon href=https://github.com/AndreaRubbi/Pear-EBI/raw/pear_ebi/logos/LOGO_PEAR.png><meta name=generator content="mkdocs-1.4.2, mkdocs-material-9.0.13"><title>1. How to use Pear - PEAR-EBI</title><link rel=stylesheet href=../assets/stylesheets/main.ffa9267a.min.css><link rel=stylesheet href=../assets/stylesheets/palette.a0c5b2b5.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=../css/print-site-enum-headings1.css><link rel=stylesheet href=../css/print-site-enum-headings2.css><link rel=stylesheet href=../css/print-site-enum-headings3.css><link rel=stylesheet href=../css/print-site-enum-headings4.css><link rel=stylesheet href=../css/print-site-enum-headings5.css><link rel=stylesheet href=../css/print-site-enum-headings6.css><link rel=stylesheet href=../css/print-site.css><link rel=stylesheet href=../css/print-site-material.css><link rel=stylesheet href=../css/timeago.css><link rel=stylesheet href=../assets/_mkdocstrings.css><link rel=stylesheet href=../docs_assets/stylesheets/pdf-export.css><link rel=stylesheet href=../css/ansi-colours.css><link rel=stylesheet href=../css/pandas-dataframe.css><script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=preference data-md-color-primary=green data-md-color-accent=indigo> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#1-how-to-use-pear class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../index.html title=PEAR-EBI class="md-header__button md-logo" aria-label=PEAR-EBI data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M392.8 1.2c-17-4.9-34.7 5-39.6 22l-128 448c-4.9 17 5 34.7 22 39.6s34.7-5 39.6-22l128-448c4.9-17-5-34.7-22-39.6zm80.6 120.1c-12.5 12.5-12.5 32.8 0 45.3l89.3 89.4-89.4 89.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l112-112c12.5-12.5 12.5-32.8 0-45.3l-112-112c-12.5-12.5-32.8-12.5-45.3 0zm-306.7 0c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3l112 112c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256l89.4-89.4c12.5-12.5 12.5-32.8 0-45.3z"/></svg> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> PEAR-EBI </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> 1. How to use Pear </span> </div> </div> </div> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/AndreaRubbi/Pear-EBI title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> PEAR-EBI </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../index.html title=PEAR-EBI class="md-nav__button md-logo" aria-label=PEAR-EBI data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M392.8 1.2c-17-4.9-34.7 5-39.6 22l-128 448c-4.9 17 5 34.7 22 39.6s34.7-5 39.6-22l128-448c4.9-17-5-34.7-22-39.6zm80.6 120.1c-12.5 12.5-12.5 32.8 0 45.3l89.3 89.4-89.4 89.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l112-112c12.5-12.5 12.5-32.8 0-45.3l-112-112c-12.5-12.5-32.8-12.5-45.3 0zm-306.7 0c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3l112 112c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256l89.4-89.4c12.5-12.5 12.5-32.8 0-45.3z"/></svg> </a> PEAR-EBI </label> <div class=md-nav__source> <a href=https://github.com/AndreaRubbi/Pear-EBI title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> PEAR-EBI </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../index.html class=md-nav__link> Home </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2> <label class=md-nav__link for=__nav_2 id=__nav_2_label tabindex=0> Modules <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_2_label aria-expanded=false> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Modules </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/gendoc.html class=md-nav__link> Gendoc </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_2> <label class=md-nav__link for=__nav_2_2 id=__nav_2_2_label tabindex=0> Pear ebi <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=2 aria-labelledby=__nav_2_2_label aria-expanded=false> <label class=md-nav__title for=__nav_2_2> <span class="md-nav__icon md-icon"></span> Pear ebi </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/pear_ebi/tree_emb_parser.html class=md-nav__link> Tree emb parser </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/tree_set.html class=md-nav__link> Tree set </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_2_3> <label class=md-nav__link for=__nav_2_2_3 id=__nav_2_2_3_label tabindex=0> Calculate distances <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=3 aria-labelledby=__nav_2_2_3_label aria-expanded=false> <label class=md-nav__title for=__nav_2_2_3> <span class="md-nav__icon md-icon"></span> Calculate distances </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/pear_ebi/calculate_distances/hashrf.html class=md-nav__link> Hashrf </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/calculate_distances/maple_RF.html class=md-nav__link> maple RF </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/calculate_distances/tqdist.html class=md-nav__link> Tqdist </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_2_4> <label class=md-nav__link for=__nav_2_2_4 id=__nav_2_2_4_label tabindex=0> Embeddings <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=3 aria-labelledby=__nav_2_2_4_label aria-expanded=false> <label class=md-nav__title for=__nav_2_2_4> <span class="md-nav__icon md-icon"></span> Embeddings </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/pear_ebi/embeddings/Isomap_e.html class=md-nav__link> Isomap e </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/embeddings/LLE_e.html class=md-nav__link> LLE e </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/embeddings/PCoA_e.html class=md-nav__link> PCoA e </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/embeddings/emb_quality.html class=md-nav__link> Emb quality </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/embeddings/tSNE_e.html class=md-nav__link> tSNE e </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_2_4_6> <label class=md-nav__link for=__nav_2_2_4_6 id=__nav_2_2_4_6_label tabindex=0> Graph <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=4 aria-labelledby=__nav_2_2_4_6_label aria-expanded=false> <label class=md-nav__title for=__nav_2_2_4_6> <span class="md-nav__icon md-icon"></span> Graph </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/pear_ebi/embeddings/graph/graph.html class=md-nav__link> Graph </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_2_5> <label class=md-nav__link for=__nav_2_2_5 id=__nav_2_2_5_label tabindex=0> Interactive mode <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=3 aria-labelledby=__nav_2_2_5_label aria-expanded=false> <label class=md-nav__title for=__nav_2_2_5> <span class="md-nav__icon md-icon"></span> Interactive mode </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/pear_ebi/interactive_mode/interactive.html class=md-nav__link> Interactive </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_2_6> <label class=md-nav__link for=__nav_2_2_6 id=__nav_2_2_6_label tabindex=0> Subsample <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=3 aria-labelledby=__nav_2_2_6_label aria-expanded=false> <label class=md-nav__title for=__nav_2_2_6> <span class="md-nav__icon md-icon"></span> Subsample </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/pear_ebi/subsample/subsample.html class=md-nav__link> Subsample </a> </li> <li class=md-nav__item> <a href=../modules/pear_ebi/subsample/subsample_multiprocess.html class=md-nav__link> Subsample multiprocess </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_3> <label class=md-nav__link for=__nav_2_3 id=__nav_2_3_label tabindex=0> Test <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=2 aria-labelledby=__nav_2_3_label aria-expanded=false> <label class=md-nav__title for=__nav_2_3> <span class="md-nav__icon md-icon"></span> Test </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../modules/test/test.html class=md-nav__link> Test </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_3 checked> <label class=md-nav__link for=__nav_3 id=__nav_3_label tabindex=0> Tutorials <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_3_label aria-expanded=true> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Tutorials </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> 1. How to use Pear <span class="md-nav__icon md-icon"></span> </label> <a href=1How_to_use_pear_ebi.html class="md-nav__link md-nav__link--active"> 1. How to use Pear </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#setup class=md-nav__link> Setup </a> <nav class=md-nav aria-label=Setup> <ul class=md-nav__list> <li class=md-nav__item> <a href=#optional class=md-nav__link> Optional </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#basic-use class=md-nav__link> Basic Use </a> <nav class=md-nav aria-label="Basic Use"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#load-trees class=md-nav__link> Load trees </a> </li> <li class=md-nav__item> <a href=#compute-distances class=md-nav__link> Compute Distances </a> </li> <li class=md-nav__item> <a href=#embed-distances class=md-nav__link> Embed distances </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#tree_set-vs-set_collection class=md-nav__link> tree_set vs set_collection </a> </li> <li class=md-nav__item> <a href=#interactive-sessions class=md-nav__link> Interactive Sessions </a> </li> <li class=md-nav__item> <a href=#there-is-more class=md-nav__link> There is more... </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=2Advanced%20use%20of%20pear_ebi.html class=md-nav__link> 2. Advanced uses of Pear </a> </li> <li class=md-nav__item> <a href=3How%20to%20use%20pear_ebi%20on%20python%20and%20reproduce%20the%20examples%20on%20the%20paper.html class=md-nav__link> 3. How to use Pear on python </a> </li> </ul> </nav> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#setup class=md-nav__link> Setup </a> <nav class=md-nav aria-label=Setup> <ul class=md-nav__list> <li class=md-nav__item> <a href=#optional class=md-nav__link> Optional </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#basic-use class=md-nav__link> Basic Use </a> <nav class=md-nav aria-label="Basic Use"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#load-trees class=md-nav__link> Load trees </a> </li> <li class=md-nav__item> <a href=#compute-distances class=md-nav__link> Compute Distances </a> </li> <li class=md-nav__item> <a href=#embed-distances class=md-nav__link> Embed distances </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#tree_set-vs-set_collection class=md-nav__link> tree_set vs set_collection </a> </li> <li class=md-nav__item> <a href=#interactive-sessions class=md-nav__link> Interactive Sessions </a> </li> <li class=md-nav__item> <a href=#there-is-more class=md-nav__link> There is more... </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <h1 id=1-how-to-use-pear>1. How to use Pear<a class=headerlink href=#1-how-to-use-pear title="Permanent link">&para;</a></h1> <p>This tutorial goes through the basic features of pear_ebi and how to use them</p> <hr> <h2 id=setup>Setup<a class=headerlink href=#setup title="Permanent link">&para;</a></h2> <p>First of all, check that you have a python version that is supported (python 3.7, 3.8, 3.9). We strongly encourage the creation of a dedicated virtual environment in order to avoid potential conflicts with other libraries due to the mismatch of dependencies' versions. We also support the use of mamba as a more efficient version of conda.</p> <p><b>In your terminal:</b><br><br> Download and install mamba: <br> <a href=https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html>See documentation</a> <br>Create new environment with one of the supported versions of python:<br> <code>mamba create -n pear_env python=3.9</code> <br>Activate environment:<br> <code>mamba activate pear_ev</code> <br>Install pear:<br> <code>python -m pip install pear_ebi</code></p> <p><br><font color=red><b>Alternatively:</b></font><br> <br>Install virtualenv:<br> <code>pip install virtualenv</code> <br>Create new environment with one of the supported versions of python:<br> <code>python3.9 -m venv pear_env</code> <br>Activate environment:<br> <code>source pear_env/bin/activate</code> <br>Install pear:<br> <code>python -m pip install pear_ebi</code></p> <h3 id=optional>Optional<a class=headerlink href=#optional title="Permanent link">&para;</a></h3> <p>If you're planning on performing more advanced analyses, such as the ones described in the <b><i>"Advanced Examples"</b></i>, you should install the extended requirements:<br> <code>python -m pip intall -r requirements.txt</code> <br>and also install the new jupyter kernel:<br> <code>python -m ipykernel install --user --name=pear_ebi</code></p> <hr> <h2 id=basic-use>Basic Use<a class=headerlink href=#basic-use title="Permanent link">&para;</a></h2> <p>After following the steps above to set up pear_ebi, you should be ready to use all the features of pear_ebi! <br>This notebook is a good guide to learn how to use it and to check that your installation is succesfull. If you should have any problem, please contact us by filing an issue on github.<br> To start with, simply check your installation by running:</p> <div class=highlight><pre><span></span><code><span class=err>!</span><span class=n>pear_ebi</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[31mNo files specified[0m[37m [0m[1;37m([0m[37msee --help for instructions[0m[1;37m)[0m
[93m- Leaving PEAR -[0m
</code></pre></div> <p>Pear is complaining because no file was given... since it looks like you don't know how to use it, it kindly suggests to seek help using the <code>--help</code> (or simply <code>-h</code>) flag. Good idea!</p> <div class=highlight><pre><span></span><code><span class=err>!</span><span class=n>pear_ebi</span> <span class=o>-</span><span class=n>h</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>usage: PEAR [-h] [-o output] [--interactive] [-d distance_matrix]
            [--meta metadata] [-m METHOD] [--pcoa PCOA] [--tsne TSNE] [--plot]
            [--config CONFIG] [--quality] [--dir DIR] [--pattern PATTERN]
            [input ...]

PEAR-EBI v0.1.85 | Phylogeny Embedding and Approximate Representation
Calculates Robison-Foulds distances between large set of trees

positional arguments:
  input                 input file : tree set in newic format

optional arguments:
  -h, --help            show this help message and exit
  -o output             output file : storage of distance matrix
  --interactive, -i     run the program in interactive mode - only the input
                        file, distance matrix, output file, and metadata
                        arguments will be considered
  -d distance_matrix, --dM distance_matrix
                        distance matrix : specify file containing a
                        precomputed distance matrix
  --meta metadata       metadata : csv file with metadata for each tree
  -m METHOD, --method METHOD
                        calculates tree distances using specified method
                        (hashrf_RF, hashrf_wRF, smart_RF, tqdist_quartet,
                        tqdist_triplet)
  --pcoa PCOA           embedding using PCoA: select number of components
                        (int) to be calculated
  --tsne TSNE           embedding using t-SNE: select number of components
                        (int) to be calculated
  --plot, -p            plot embedding in 2 or 3 dimensions
  --config CONFIG, -c CONFIG
                        toml config file
  --quality, -q         asess quality of embedding
  --dir DIR             directory with files
  --pattern PATTERN     pattern of files in directory

Author: Andrea Rubbi - Goldman Group | European Bioinformatics Institute
</code></pre></div> <p>Essentially, pear_ebi does the following:<ul> <li>Reads phylogenetic trees in newick format from one or multiple files; <li>Computes the distances between the trees with one of the available metrics (refer to <code>--help</code> to see them and to the manuscript for their specific functions); <li>Embeds the distances in a lower dimensional space. Basic usage allows using either PCoA or tSNE; <li>Plots the resulting embeddings. </ul> Why should you use pear_ebi? Because it's <font color=green><b>simple</b><font color=black>, <font color="dark orange"><b>fast</b><font color=black>, and <font color=purple><b>produces nice graphs</b></font>! </p> <h3 id=load-trees>Load trees<a class=headerlink href=#load-trees title="Permanent link">&para;</a></h3> <p>Pear has two kind of structures for your trees:<ol> <li><font color=blue>tree_set</font>;<li><font color="dark pink">set_collection</font>. </ol> You define a <font color=blue>tree_set</font> with a set of trees coming from a single file, whereas a <font color="dark pink">set_collection</font> is composed of trees coming from multiple files, divided into groups depending on the file of origin. In practice, computationally and conceptually, a <font color="dark pink">set_collection</font> is a set of multiple <font color=blue>tree_set</font>. <br>We'll make it clear exploring how we can load trees on pear with a few examples.</p> <div class=highlight><pre><span></span><code><span class=c1># load single set of trees into a tree_set #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>beast_trees</span><span class=o>/</span><span class=n>beast_long</span><span class=o>.</span><span class=n>trees</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m1001[0m trees;
 File: beast_trees/beast_long.trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€


[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># load two set of trees into a set_collection #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>beast_trees</span><span class=o>/</span><span class=n>beast_run1</span><span class=o>.</span><span class=n>trees</span> <span class=n>beast_trees</span><span class=o>/</span><span class=n>beast_run2</span><span class=o>.</span><span class=n>trees</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€            
 Tree set collection containing [1;36m2002[0m trees;            
 File: Set_collection_[93m1bfec55f-592e-43a6-bf2b-4000219b3092[0m;
 Distance matrix: not computed.                
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ 
beast_run1; Containing [1;36m1001[0m trees. 
beast_run2; Containing [1;36m1001[0m trees.


[93m- Leaving PEAR -[0m
</code></pre></div> <p>Although you may be perplexed about the actual practical difference between these structures, we suggest you keep these questions for the next chapters and bear with us to explore some more funky ways of loading trees onto pear. <br>In fact, you can avoid boring repetitions of the path of your files and simply indicate the directory! </p> <div class=highlight><pre><span></span><code><span class=c1># load files from directory #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=o>--</span><span class=nb>dir</span> <span class=n>beast_trees</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€            
 Tree set collection containing [1;36m9009[0m trees;            
 File: Set_collection_[93m696e3bb7-7deb-46e1-97c8-3bcce630cddd[0m;
 Distance matrix: not computed.                
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ 
beast_run7; Containing [1;36m1001[0m trees. 
beast_run2; Containing [1;36m1001[0m trees. 
beast_long; Containing [1;36m1001[0m trees. 
beast_run5; Containing [1;36m1001[0m trees. 
beast_run6; Containing [1;36m1001[0m trees. 
beast_run3; Containing [1;36m1001[0m trees. 
beast_run8; Containing [1;36m1001[0m trees. 
beast_run4; Containing [1;36m1001[0m trees. 
beast_run1; Containing [1;36m1001[0m trees.


[93m- Leaving PEAR -[0m
</code></pre></div> <p>Whoa! Too many... thankfully we can indicate a pattern to look for</p> <div class=highlight><pre><span></span><code><span class=c1># load files from directory using pattern#</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=o>--</span><span class=nb>dir</span> <span class=n>beast_trees</span> <span class=o>--</span><span class=n>pattern</span> <span class=s2>&quot;*run*&quot;</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€            
 Tree set collection containing [1;36m8008[0m trees;            
 File: Set_collection_[93m9781a60f-d6d7-4ffb-97c3-36d7aed9af4b[0m;
 Distance matrix: not computed.                
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ 
beast_run7; Containing [1;36m1001[0m trees. 
beast_run2; Containing [1;36m1001[0m trees. 
beast_run5; Containing [1;36m1001[0m trees. 
beast_run6; Containing [1;36m1001[0m trees. 
beast_run3; Containing [1;36m1001[0m trees. 
beast_run8; Containing [1;36m1001[0m trees. 
beast_run4; Containing [1;36m1001[0m trees. 
beast_run1; Containing [1;36m1001[0m trees.


[93m- Leaving PEAR -[0m
</code></pre></div> <p>If you are a regex wizard you can probably select any set of similarly-named files in this way. Me? I generally ask GPT to write the magic formula. <b>NB:</b> you can also use a combination of <code>--dir</code>, <code>--pattern</code>, and normal file definition if one or more files come from other directories. </p> <div class=highlight><pre><span></span><code><span class=err>!</span><span class=n>pear_ebi</span> <span class=n>beast_trees</span><span class=o>/</span><span class=n>beast_long</span><span class=o>.</span><span class=n>trees</span> <span class=o>--</span><span class=nb>dir</span> <span class=n>beast_trees</span> <span class=o>--</span><span class=n>pattern</span> <span class=s2>&quot;*run[1,2]*&quot;</span> 
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€            
 Tree set collection containing [1;36m3003[0m trees;            
 File: Set_collection_[93m21d56458-1d6b-4dab-9842-f8a65287e845[0m;
 Distance matrix: not computed.                
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ 
beast_run2; Containing [1;36m1001[0m trees. 
beast_run1; Containing [1;36m1001[0m trees. 
beast_long; Containing [1;36m1001[0m trees.


[93m- Leaving PEAR -[0m
</code></pre></div> <h3 id=compute-distances>Compute Distances<a class=headerlink href=#compute-distances title="Permanent link">&para;</a></h3> <p>You can compute the distance matrix using different methods. Each method has a specific purpose, which is outlined in the associated paper. However, in general, the Robison Foulds distance metric is an admittedly good choice. Additionally, pear_ebi computes this metric using the hashrf algorithm, which is the fastest way of computing such metric to date. You can indicate any method available using the <code>-m</code>, or <code>--method</code>, flag. When a metric/method is indicated, pear will use it to compute the distance matrix <b>even if a distance matrix is given</b>, and it <b>will overscribe any previous matrix saved at a file with the same standard format</b> (realistically, this happens only if the matrix was produced using pear during an interactive or advanced session - see the interactive sessions chapter).</p> <div class=highlight><pre><span></span><code><span class=c1># compute Robison Foulds distances using hashrf #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>beast_trees</span><span class=o>/</span><span class=n>beast_run1</span><span class=o>.</span><span class=n>trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>hashrf_RF</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m1001[0m trees;
 File: beast_trees/beast_run1.trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ™[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># compute Robison Foulds distances using hashrf #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=o>--</span><span class=nb>dir</span> <span class=n>beast_trees</span> <span class=o>--</span><span class=n>pattern</span> <span class=s2>&quot;*run[12]*&quot;</span> <span class=o>--</span><span class=n>method</span> <span class=n>hashrf_RF</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€            
 Tree set collection containing [1;36m2002[0m trees;            
 File: Set_collection_[93mc9cf5472-d392-4825-b88f-e9feffdb331a[0m;
 Distance matrix: not computed.                
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ 
beast_run2; Containing [1;36m1001[0m trees. 
beast_run1; Containing [1;36m1001[0m trees.

[2K[32mâ ™[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># compute quartet distances #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>tqdist_quartet</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ´[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mtqdist_quartet | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># compute modified RF distances #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>smart_RF</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ™[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34msmart_RF | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <p>If you run the exaples above you will notice the difference between the time efficiency of hashrf and any other algorithm. Please note the astonishing performance of that algorithm, especially considering the sensible difference in the number of trees analyzed!</p> <p>Given that one may want to analyze the distance matrix in multiple ways, possibly desiring to skip the hefty distance-computation step, we introduced the convenient <code>-d</code> flag that allows specifying a previosuly computed distance matrix. <b>Please note that</b> the order of the trees in the distance matrix is preserved and thus one must be consistent with the input specification when reusing a distance matrix. Should you disregard this suggestion, we suggest you also disregard your downstream analyses (and perhaps everything else as well).</p> <div class=highlight><pre><span></span><code><span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>hashrf_RF</span> <span class=o>-</span><span class=n>o</span> <span class=n>precomputed_distance_matrix</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ´[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># reusing distances #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>d</span> <span class=n>precomputed_distance_matrix</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€


[93m- Leaving PEAR -[0m
</code></pre></div> <p>Please note the difference in the input "Distance matrix" status.</p> <h3 id=embed-distances>Embed distances<a class=headerlink href=#embed-distances title="Permanent link">&para;</a></h3> <p>Here we show the easy way of embedding the distances and produce nice plots! First of all, one may choose any number of dimensions <span class=arithmatex>\(M \leq N\)</span>; where <span class=arithmatex>\(N\)</span> is the number of trees and <span class=arithmatex>\(M\)</span> the chosen dimensionality of the embedded space. Note that, should <span class=arithmatex>\(M\)</span> be <span class=arithmatex>\(\lt 3\)</span>, pear will produce only one 2D plot. Whereas, should <span class=arithmatex>\(M\)</span> be <span class=arithmatex>\(\geq 3\)</span> pear will produce a 2D and a 3D plot. Please note that we are currently unable to produce human-friendly representations for <span class=arithmatex>\(\gt 3\)</span>D data... In any case, whichever dimension <span class=arithmatex>\(M\)</span> may be chosen, an <span class=arithmatex>\(M\)</span>-dimensional embedding shall be produced and then saved on the machine.</p> <div class=highlight><pre><span></span><code><span class=c1># compute triplet distances, embed in 2D using pcoa #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>tqdist_triplet</span> <span class=o>--</span><span class=n>pcoa</span> <span class=mi>2</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ¹[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mtqdist_triplet | Done![0m
[?25l[32mâ ‹[0m [1;32mEmbedding distances...[0m
[1A[2K[1;34mpcoa | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># compute RF distances, embed in 5D using tsne #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>hashrf_RF</span> <span class=o>--</span><span class=n>tsne</span> <span class=mi>5</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ¦[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m
[2K[32mâ ¸[0m [1;32mEmbedding distances...[0m0m
[1A[2K[1;34mtsne | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <p>Use the <code>--plot</code> flag to show the plots at the end. Note that, when an embedding method is specified, the plots are produced regardless of whether the <code>--plot</code> flag is present or not. </p> <div class=highlight><pre><span></span><code><span class=c1># compute RF distances, embed in 5D using tsne #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>hashrf_RF</span> <span class=o>--</span><span class=n>tsne</span> <span class=mi>5</span> <span class=o>--</span><span class=n>plot</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ´[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m
[2K[32mâ ¸[0m [1;32mEmbedding distances...[0m0m
[1A[2K[1;34mtsne | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <p><b>NB:</b> this is something you are expected to run in a terminal, this is why the graph doesn't show up here.</p> <p>The <code>--quality</code>, or <code>-q</code>, flag indicates to pear to provide some quality metrics for the embedding generated. <br> The toolset of quality-measures provided by pear may be expanded upon request in the future.</p> <div class=highlight><pre><span></span><code><span class=c1># compute RF distances, embed in 5D using tsne, computing quality #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>hashrf_RF</span> <span class=o>--</span><span class=n>tsne</span> <span class=mi>5</span> <span class=o>--</span><span class=n>quality</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ¼[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m
[2K[32mâ ¼[0m [1;32mEmbedding distances...[0m0m
[1A[2K[1;34mtsne | Done![0m
With [1;36m5[0m components/dimensions, the estimated correlation with the [1;36m32[0m-dimensional 
coordinates is [1;36m-0.03[0m

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># compute RF distances, embed in 2D using pcoa, computing quality #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>hashrf_RF</span> <span class=o>--</span><span class=n>pcoa</span> <span class=mi>2</span> <span class=o>--</span><span class=n>quality</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ¼[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m
[?25l[32mâ ‹[0m [1;32mEmbedding distances...[0m
[1A[2K[1;34mpcoa | Done![0m
With [1;36m2[0m components/dimensions, the explained variance is [1;36m96.56[0m,
 with an estimated correlation [1;36m1.00[0m with the [1;36m32[0m-dimensional coordinates

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=c1># compute RF distances, embed in 10D using pcoa, computing quality #</span>
<span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>m</span> <span class=n>hashrf_RF</span> <span class=o>--</span><span class=n>pcoa</span> <span class=mi>10</span> <span class=o>--</span><span class=n>quality</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ¦[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m
[?25l[32mâ ‹[0m [1;32mEmbedding distances...[0m
[1A[2K[1;34mpcoa | Done![0m
With [1;36m10[0m components/dimensions, the explained variance is [1;36m99.98[0m,
 with an estimated correlation [1;36m1.00[0m with the [1;36m32[0m-dimensional coordinates

[93m- Leaving PEAR -[0m
</code></pre></div> <p><i>Pro tip</i>: since in order to compute an embedded space we need the distance matrix first, pear will automatically compute the distances with <code>hashrf_RF</code> when no method is indicated (and no precomputed distance matrix is specified) </p> <div class=highlight><pre><span></span><code><span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>--</span><span class=n>pcoa</span> <span class=mi>2</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[2K[32mâ ¦[0m [1;32mCalculating distances...[0m0m
[1A[2K[1;34mhashrf_RF | Done![0m
[?25l[32mâ ‹[0m [1;32mEmbedding distances...[0m
[1A[2K[1;34mpcoa | Done![0m

[93m- Leaving PEAR -[0m
</code></pre></div> <h2 id=tree_set-vs-set_collection><font color=blue>tree_set</font> vs <font color="dark pink">set_collection</font><a class=headerlink href=#tree_set-vs-set_collection title="Permanent link">&para;</a></h2> <p>Pear has two kind of structures for your trees:<ol> <li><font color=blue>tree_set</font>;<li><font color="dark pink">set_collection</font>. </ol> <font color=blue>tree_set</font> allows for analyzing a single set of trees. <font color="dark pink">set_collection</font> allows for analyzing multiple set of trees and the divergences between them. In our interactive plots, <font color="dark pink">set_collection</font> generates different sets of points that are colored according to the file of origin. This could be useful, for instance, if one performs multiple runs of a phylogenetic-tree estimating algorithm and wants to assess the robustness/consistency of that method. <br>The distances computation is overall the same as a square distance matrix is generated, encompassing the whole collection of trees.</p> <hr> <h2 id=interactive-sessions>Interactive Sessions<a class=headerlink href=#interactive-sessions title="Permanent link">&para;</a></h2> <p>All the above, but staying in the loop! <br>You can add set of trees iteratively and compute distances/embeddings as many times as you want to. </p> <div class=highlight><pre><span></span><code><span class=err>!</span><span class=n>pear_ebi</span> <span class=o>-</span><span class=n>i</span>
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mSpecify file with tree set[0m

File: ^C

[93m- Leaving PEAR -[0m
</code></pre></div> <div class=highlight><pre><span></span><code><span class=err>!</span><span class=n>pear_ebi</span> <span class=n>MAPLE_res</span><span class=o>/</span><span class=n>IQtreeStartingTree_slower_Trees</span> <span class=o>-</span><span class=n>i</span> 
</code></pre></div> <div class=codehilite><pre><span></span><code>[34mPEAR v0.[0m[1;34m1.85[0m
[95mYour input:[0m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 Tree set containing [1;36m32[0m trees;
 File: MAPLE_res/IQtreeStartingTree_slower_Trees;
 Distance matrix: not computed.
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

[1;32mPEAR | Interactive Mode[0m
[32mâ£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â ¿â£Ÿâ ‰â¡¿â ¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿[0m[1;37m -- Controls --[0m
[32mâ£¿â£¿â£¿â£¿â¡¿â£¿â¢‰â¢³â ´â£žâ ‰â¡·â¢¥â¡â¡™â¡¿â¢¿â£¿â£¿â£¿[0m[1;37m [0m[1;37m1[0m[1;37m --&gt; see status[0m
[32mâ£¿â£¿â¡‹â¢»â¡¤â£¼â ‰â¢¯â¡†â£žâ ™â£§â£¢â â ªâ££â¢¦â¡›â ¹â£¿[0m[1;37m [0m[1;37m2[0m[1;37m --&gt; calculate distances[0m
[32mâ£¿â£¿â “â¢»â£„â£¼â ‹â¢·â¡ â¡½â šâ£‰â£¤â¡žâ¢šâ¢¦â¢¢â Ÿâ ¹â£¿[0m[1;37m [0m[1;37m3[0m[1;37m --&gt; embed distances[0m
[32mâ£¿â£¿â¢“â¢»â¡„â¡¼â —â¢ƒâ£‚â¡’â »â£§â£‚â¡¿â šâ£¨â¢¨â¡“â »â£¿[0m[1;37m [0m[1;37m4[0m[1;37m --&gt; plot embeddings[0m
[32mâ£¿â£¿â —â¢Žâ¢„â ‚â ¾â£¯â£‚â¡½â¢“â¢†â¢”â â ¿â£‡â¢…â¡—â »â£¿[0m[1;37m [0m[1;37m5[0m[1;37m --&gt; add set to collection[0m
[32mâ£¿â£¿â –â¢¯â¡¡â£¹â —â£¤â£‰â ›â ¶â£â¢Œâ¡¿â ²â¡Œâ¢Œâ¢žâ ¼â£¿[0m[1;37m [0m[1;37m6[0m[1;37m --&gt; get subset[0m
[32mâ£¿â£¿â£®â£¾â¡‰â£¹â ¦â£žâ¡‰â¡½â ¶â¡Œâ Œâ¡®â ²â£â¢©â£·â£¼â£¿[0m
[32mâ£¿â£¿â£¿â£¿â£¿â£¿â£¤â£žâ¢‰â£³â ¥â£â â£§â£µâ£¿â£¿â£¿â£¿â£¿[0m[1;37m [0m[1;37m7[0m[1;37m --&gt; exit[0m
[32mâ£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£·â£¤â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿[0m[1;37m [0m[1;37m8[0m[1;37m --&gt; see list of controls[0m
Command:
</code></pre></div> <h2 id=there-is-more>There is more...<a class=headerlink href=#there-is-more title="Permanent link">&para;</a></h2> <p>the flags <code>--meta</code> and <code>--config</code> introduce another layer of flexibility into our analyses. <br> Please, refer to the <b><i>"Advanced Examples"</b></i> folder to get a gist of the full potential of pear!</p> <hr> <div class=md-source-file> <small> Last update: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago"><span class=timeago datetime=2024-04-29T10:13:54+00:00 locale=en></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date">2024-04-29</span> </small> </div> </article> </div> </div> </main> <footer class=md-footer> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Made by Andrea Rubbi </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=https://www.linkedin.com/in/andrea-rubbi-9a06021a1/ target=_blank rel=noopener title=www.linkedin.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script> <script src=../assets/javascripts/bundle.2a6f1dda.min.js></script> <script src=../js/print-site.js></script> <script src=../js/timeago.min.js></script> <script src=../js/timeago_mkdocs_material.js></script> <script src=../javascripts/config.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>